lib/renderer/api/ipc-renderer.js

```diff
@@ -27,4 +27,11 @@ ipcRenderer.sendToAll = function (webContentsId, channel, ...args) {
   return ipc.sendTo(internal, true, webContentsId, channel, args)
 }
 
+ipcRenderer.invoke = function (channel, ...args) {
+  return ipc.invoke(channel, args).then(({ error, result }) => {
+    if (error) { throw new Error(`Error invoking remote method '${channel}': ${error}`) }
```

Comment:
use `deserialize` from `lib/common/error-utils.js`?

Reply:
Hmmm, I'm not sure we want to re-throw the same error that was emitted in the main process.

1. When the error is thrown in the renderer process, it's important to know that the error came _from an invoke() call_, which `errorUtils.deserialize` masks. Otherwise it could be very confusing to see a stack trace from the main process in the renderer.
2. JavaScript in the renderer process shouldn't have access to details about the main process by default. Stack traces can contain potentially sensitive information. I'm cautious about the security implications of transmitting stack data from the main process to the renderer.

If this was Java I'd suggest we use [exception causes](https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html) but JS doesn't seem to have an equivalent.

Reply:
I agree the stack shouldn't be sent, but it should still be traceable IMO.

E.g. Log the error in the main process with an error ID, send the error ID to the renderer so that is can say "Error 1234 in the main process logs"

Reply:
Hm... yeah, I agree that the error stack should be visible _somewhere_. The error ID thing is neat but is kind of a junk developer experience.

Ideally someone would convince me that passing back `e.stack` info to `invoke` callers is O.K. security wise (since e.g. sandbox / contextIsolation should be preventing the data from leaking). And then we could add the stack trace in after the existing stack, something like `e.stack += '\nCaused by: ' + remoteError.stack`

Reply:
I settled on a compromise for now: we log the error in the main process without any IDs to match. 95% of the time I think it will be obvious which error in the main process matches with which error in the renderer process (we print the channel and the error message in both main & renderer).